src/utp_socket_manager.cpp

```diff
@@ -255,12 +255,13 @@ namespace libtorrent
 
 	void utp_socket_manager::writable()
 	{
-		std::vector<utp_socket_impl*> stalled_sockets;
+		if (m_stalled_sockets.empty()) return;
+
+		static socket_vector_t stalled_sockets;
+		stalled_sockets.clear();
 		m_stalled_sockets.swap(stalled_sockets);
-		for (std::vector<utp_socket_impl*>::iterator i = stalled_sockets.begin()
-			, end(stalled_sockets.end()); i != end; ++i)
+		for (auto const &s : stalled_sockets)
 		{
-			utp_socket_impl* s = *i;
 			utp_writable(s);
```

Comment:
perhaps one of these functions are "reentrant". i.e. utp_socket::writable may be called while calling a handler.